### PR TITLE
Refactor `RestClient` for unit test

### PR DESCRIPTION
**Pull request by Y-Sindo**
- Refactor `RestClient` for unit test

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Common/Utilities/HttpClientFactory.cs [hunk 1]
```diff
@@ -6,22 +6,10 @@
...
-            var services = serviceCollection.BuildServiceProvider();
-
-            _httpClientFactory = services.GetRequiredService<IHttpClientFactory>();
-        }
-
-        public static HttpClient CreateClient()
-        {
-            return _httpClientFactory.CreateClient();
-        }
+        public static IHttpClientFactory Instance { get; } = new ServiceCollection().AddHttpClient()
```
Comment (wanlwanl) [inline src/Microsoft.Azure.SignalR.Common/Utilities/HttpClientFactory.cs:11]: no need to manually create it from a new ServiceCollection

Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Common/Utilities/HttpClientFactory.cs:11]: Currently `AadAccessKey` need `new RestClient()` and this static instance. @terencefan

File: src/Microsoft.Azure.SignalR.Common/Utilities/RestClient.cs [hunk 2]
```diff
@@ -45,7 +58,7 @@ internal class RestClient
             Func<HttpResponseMessage, Task<bool>> handleExpectedResponseAsync = null,
             CancellationToken cancellationToken = default)
         {
-            var httpClient = HttpClientFactory.CreateClient();
+            var httpClient = _httpClientFactory.CreateClient();
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.Common/Utilities/RestClient.cs:61]: using

File: src/Microsoft.Azure.SignalR.Common/Utilities/RestClient.cs [hunk 3]
```diff
@@ -45,32 +58,34 @@ internal class RestClient
             Func<HttpResponseMessage, Task<bool>> handleExpectedResponseAsync = null,
             CancellationToken cancellationToken = default)
         {
-            var httpClient = HttpClientFactory.CreateClient();
+            using var httpClient = _httpClientFactory.CreateClient();
             var request = BuildRequest(api, httpMethod, productInfo, methodName, args);
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.Common/Utilities/RestClient.cs:62]: `using var ...`

**Review by vwxyzh — APPROVED** [review]

**Pull request merged**